src/StreamJsonRpc.Tests/JsonRpcProxyGenerationTests.cs

```diff
@@ -331,6 +331,42 @@ public void InternalInterface()
         Assert.False(tcs.Task.IsCompleted);
     }
 
+    [Fact]
+    public async Task NamingTransformsAreAppliedToEvents()
+    {
+        var streams = FullDuplexStream.CreateStreams();
+        this.serverStream = streams.Item1;
+        this.clientStream = streams.Item2;
+
+        var camelCaseOptions = new JsonRpcProxyOptions { MethodNameTransform = CommonMethodNameTransforms.CamelCase };
+        var prefixOptions = new JsonRpcProxyOptions { MethodNameTransform = CommonMethodNameTransforms.Prepend("ns.") };
+
+        // Construct two client proxies with conflicting method transforms to prove that each instance returned retains its unique options.
+        var clientRpc = new JsonRpc(this.clientStream, this.clientStream);
+        var clientRpcWithCamelCase = clientRpc.Attach<IServer>(camelCaseOptions);
+        var clientRpcWithPrefix = clientRpc.Attach<IServer>(prefixOptions);
+        clientRpc.StartListening();
+
+        // Construct the server to only respond to one set of method names for now to confirm that the client is sending the right one.
+        this.serverRpc = new JsonRpc(this.serverStream, this.serverStream);
+        this.serverRpc.AddLocalRpcTarget(this.server, new JsonRpcTargetOptions { MethodNameTransform = camelCaseOptions.MethodNameTransform });
+        this.serverRpc.StartListening();
+
+        var tcs = new TaskCompletionSource<EventArgs>();
+        EventHandler handler = (sender, args) => tcs.SetResult(args);
+        clientRpcWithCamelCase.ItHappened += handler;
+        this.server.OnItHappened(EventArgs.Empty);
+        var actualArgs = await tcs.Task.WithCancellation(this.TimeoutToken);
+        Assert.NotNull(actualArgs);
+
+        clientRpcWithCamelCase.ItHappened -= handler;
+        clientRpcWithPrefix.ItHappened += handler;
```

Comment:
I know it won't matter here, but I always prefer unsubscribing handlers holistically. That way if the test gets extended in the future someone doesn't miss this (so suggesting that after your final assert you add clientRpcWithPrefix.ItHappened -= handler;

Reply:
Resolved.